rust-toolchain

```diff
@@ -1 +1 @@
-1.63.0
+1.73.0
```

Comment:
Of the dependencies that this PR updates:
* `secp256k1` master has MSRV 1.48.0 (it isn't declared in its `Cargo.toml` but see https://github.com/rust-bitcoin/rust-secp256k1/tree/master/secp256k1-sys#minimum-supported-rust-version )
* `ring` main has MSRV 1.61.0: https://github.com/briansmith/ring/blob/6762d656742d7ea431948356760da28e09292b72/Cargo.toml#L14
* `thiserror` master has MSRV 1.56: https://github.com/dtolnay/thiserror/blob/0717de3f507293f6faa7a32d987febb5c39a7048/Cargo.toml#L12
* `hex` has no declared MSRV, and its only non-dev dependency is `serde` with an MSRV of 1.31.0: https://github.com/KokaKiwi/rust-hex/blob/main/Cargo.toml
* `base58` has no declared MSRV and no dependencies: https://github.com/debris/base58/blob/master/Cargo.toml
* `ripemd160` is deprecated and should be replaced with the `ripemd` crate. `ripemd 0.1.3` has MSRV 1.41: https://crates.io/crates/ripemd/0.1.3
* `rand` master has MSRV 1.60: https://github.com/rust-random/rand/blob/f827c006b52676301add803671b5d063191dcdad/Cargo.toml#L17

```suggestion
1.63.0
```